examples/key-example.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <mococrw/key.h>
+#include <iostream>
+
+using namespace mococrw;
+
+void createRsaKey(uint keySize, std::string password)
+{
+    std::shared_ptr<AsymmetricKey::Spec> spec;
+    spec = std::make_shared<RSASpec>(keySize);
```

Comment:
```suggestion
    std::shared_ptr<AsymmetricKey::Spec> spec = std::make_shared<RSASpec>(keySize);
```

Would just keep this in a single line.

Reply:
Why even have a ptr?